lib/charms/ovn_charm.py

```diff
@@ -608,6 +610,24 @@ def configure_bridges(self):
             # NOTE(fnordahl): Workaround for LP: #1848757
             opvs.remove('.', 'external_ids', 'ovn-cms-options')
 
+    def render_nrpe(self):
+        """Configure Nagios NRPE checks."""
+        ch_core.hookenv.log("Rendering NRPE checks.",
+                            level=ch_core.hookenv.INFO)
+        hostname = nrpe.get_nagios_hostname()
+        current_unit = nrpe.get_nagios_unit_name()
+        # Determine if this is a subordinate unit or not
+        if ch_core.hookenv.principal_unit() == ch_core.hookenv.local_unit():
+            primary = True
+        else:
+            primary = False
+        charm_nrpe = nrpe.NRPE(hostname=hostname, primary=primary)
+        if reactive.is_flag_set('nova-compute.available'):
+            self.nrpe_check_services.append(self.metadata_agent)
```

Comment:
The charm class `__init__` method already makes choices based on OpenStack being present or not, could we do this there instead?